lib/web/integrations.go

```diff
@@ -224,3 +228,44 @@ func (h *Handler) integrationsList(w http.ResponseWriter, r *http.Request, p htt
 		NextKey: nextKey,
 	}, nil
 }
+
+// integrationsMsTeamsAppZipGet generates and returns the app.zip required for the MsTeams plugin with the given name.
+func (h *Handler) integrationsMsTeamsAppZipGet(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (interface{}, error) {
+	clt, err := sctx.GetUserClient(r.Context(), site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	targetDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.RemoveAll(targetDir)
+
+	plugin, err := clt.PluginsClient().GetPlugin(r.Context(), &pluginspb.GetPluginRequest{
+		Name:        p.ByName("plugin"),
+		WithSecrets: false,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	spec, ok := plugin.Spec.Settings.(*types.PluginSpecV1_MsTeams)
+	if !ok {
+		return nil, trace.BadParameter("plugin specified was not of type MsTeams")
+	}
+
+	msteams.ConfigureAppZip(targetDir, "app.zip", msteams.Payload{
```

Comment:
Can you do this in a way that doesn't require you to temporarily write to disk? I'm not sure we have much space on our ephemeral container filesystems.

Reply:
+1 the filesystem might not be readable, is slower, and will leave leftovers. `msteams.ConfigureAppZip` should be updated to accept an `io.Writer` instead of a filename. Then, the webserver will be able to pass a buffer, then read the buffer content back to the HTTP response .